workspaces/adventure-pack/goodies/python3/src/list_transpose/__init__.py

```diff
@@ -0,0 +1,16 @@
+from typing import Any
+
+
+def list_transpose(list: list[Any]) -> list[Any]:
+    r = len(list)
+    c = len(list[0]) if r != 0 else 0
+    if r == 0 or c == 0:
+        return []
+    if any(len(row) != c for row in list):
+        return []
```

Comment:
This seems a little bit suspicious, shouldn't we raise an error when we can't do the requested operation? I could also be convinced that we should return None, if that's Pythonic. Curious how you thought about this.

Reply:
Raising an error is the better way of doing it, I fixed it now.